### PR TITLE
Update TcpSocket API Interface

### PR DESCRIPTION
**Pull request by dborkan**
Update TcpSocket API Interface to match new Freedom API

### CODE REVIEW

File: interfaces/tcp-socket.d.ts [hunk 1]
```diff
@@ -4,13 +4,9 @@
...
-  }
-
   export interface DisconnectInfo {
-    socketId:number;
-    error:string;
+    errcode :number;
+    message :string;
   }
 
   export interface ReadInfo {
```
Comment (trevj) [inline interfaces/tcp-socket.d.ts:12]: I think we don't need the socketId any more, right?

File: interfaces/tcp-socket.d.ts [hunk 2]
```diff
@@ -22,9 +18,6 @@ declare module freedom.TcpSocket {
     bytesWritten: number;
   }
 
-  export interface SocketOptions {
-  }
-
   export interface SocketInfo {
     socketType: string;
```
Comment (trevj) [inline interfaces/tcp-socket.d.ts]: I don't think we need socketType any more, either?

File: interfaces/tcp-socket.d.ts [hunk 3]
```diff
@@ -22,9 +18,6 @@ declare module freedom.TcpSocket {
     bytesWritten: number;
   }
 
-  export interface SocketOptions {
-  }
-
   export interface SocketInfo {
```
Comment (trevj) [inline interfaces/tcp-socket.d.ts:20]: Are the fields in here really optional?

File: interfaces/tcp-socket.d.ts [hunk 4]
```diff
@@ -36,30 +29,25 @@ declare module freedom.TcpSocket {
 
   // TODO: move to tcp socket implementation; drop the type argument.
```
Comment (iislucas) [inline interfaces/tcp-socket.d.ts]: This comment can now be removed. :)

File: interfaces/tcp-socket.d.ts [hunk 5]
```diff
@@ -36,30 +29,25 @@ declare module freedom.TcpSocket {
...
         continuation: (result: number) => void): void;
-    destroy(socketId:number, continuation:() => void): void;
-    disconnect(socketId:number, continuation:() => void): void;
+    close(continuation:() => void): void;
   }
 }  // module Sockets
 
 // The TcpSocket class (freedom['core.TcpSocket'])
 declare module freedom {
   class TcpSocket {
```
Comment (iislucas) [inline interfaces/tcp-socket.d.ts]: Please add real types (`any` is usually a placeholder) in this CL, or a TODO for it for a followup CL.

File: interfaces/tcp-socket.d.ts [hunk 6]
```diff
@@ -36,30 +29,25 @@ declare module freedom.TcpSocket {
 
   // TODO: move to tcp socket implementation; drop the type argument.
   export interface Implementation {
```
Comment (trevj) [inline interfaces/tcp-socket.d.ts]: So, you might consider doing this file in the style of social.d.ts and transport.d.ts. They just define the interface _as seen by the client_, e.g. tcp.ts, and are easier to read and don't make any mention of continuations.

Then it'll be easier to see that, for example, getInfo() takes no arguments.

**Comment by trevj** [comment]
Great to see this going in.

You'll need to update the NPM. I added some instructions to the coding guide:
https://docs.google.com/a/google.com/document/d/1rnaSOBbAHYvb9yXr8lyM-nkUe2FefzUWpeYZ27UKr7o/edit#heading=h.kd9qtmrrdpr8

(let me know if you have issues, it's very possible I've missed something)

File: interfaces/tcp-socket.d.ts (hunk 1 above)
Comment (dborkan) [inline interfaces/tcp-socket.d.ts:12]: Correct, I just removed it from ReadInfo

File: interfaces/tcp-socket.d.ts (hunk 3 above)
Comment (dborkan) [inline interfaces/tcp-socket.d.ts:20]: No, just corrected

File: interfaces/tcp-socket.d.ts (hunk 2 above)
Comment (dborkan) [inline interfaces/tcp-socket.d.ts]: Correct, removed

File: interfaces/tcp-socket.d.ts (hunk 4 above)
Comment (dborkan) [inline interfaces/tcp-socket.d.ts]: Done

File: interfaces/tcp-socket.d.ts (hunk 6 above)
Comment (dborkan) [inline interfaces/tcp-socket.d.ts]: Done

**Comment by iislucas** [comment]
LGTM

**Pull request merged**